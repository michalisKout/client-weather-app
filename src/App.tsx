import { getWeatherCurrentLocationAsync } from '@/domain/store/modules/weather';
import { AppDispatch, useAppSelector } from '@/domain/store/store.types';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

function App() {
  const dispatch = useDispatch<AppDispatch>();
  const weatherData = useAppSelector((store) => store.weather.data);

  useEffect(() => {
    dispatch(getWeatherCurrentLocationAsync({ city: 'Athens' }));
  }, []);

  return (
    <>
      <h1 className="text-red-700 text-lg">Weather app</h1>
      {weatherData?.location.country}
    </>
  );
}

export default App;
