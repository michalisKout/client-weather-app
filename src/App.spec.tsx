import App from '@/App';
import { render } from '@testing-library/react';

describe('App', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<App />);
    expect(true).toBe(true);

    expect(baseElement).toMatchSnapshot();
  });
});
