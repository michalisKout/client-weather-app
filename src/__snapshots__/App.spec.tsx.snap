// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match snapshot 1`] = `
<body>
  <div>
    <h1>
      Weather app
    </h1>
  </div>
</body>
`;
